2020/09/02/rclone-upload-to-cloud-from-cli/rclone-upload-to-cloud-from-cli.html.md

```diff
@@ -0,0 +1,239 @@
+
+```bash
+for i in $( ls /drives/source/directory ) ;
+do gpg --output /mnt/rclone/drive/$i.gpg --recipient recipient@email.com --encrypt /drives/source/directory/$i
+done
+```
+
+Quite a bit of setup. But the result is a mounted folder into which you can copy encrypted files into directly, without any intermediary steps. The above command took a while, but by automating the task, the total work time was still less than it would have been trying to do it manually- especially given that we didn’t have the spare disk space to encrypt the drives in to.
+
+Want to do more? You could take a whole drive image with dd, run it through gzip and gpg, and pipe the output to the remote storage. You could use the crypt module for your own backups, rather than relying on gpg. The mount point acts as though it was a drive on your system, so any system utility could back up to the drive- Timeshift, for instance.
+
+If you have any other suggestions for how this might be used, please leave a comment!
+
+### DIY Google API key
+*These instructions can also be found on the [rclone website]*
+
+For this blog post I set this up using my personal account and went through the steps below- some of the steps may be different if you use a G-Suites account. It's a little cumbersome but will only take 5-10 minutes at most.
+
+Log in to the Developers console: https://console.developers.google.com/ - then create a new project.
+
+<a href="/2020/09/02/Rclone-upload-to-the-cloud-from-cli/rclone_google_api_1-full.png"><img src="/2020/09/02/Rclone-upload-to-the-cloud-from-cli/rclone_google_api_1-full.png" alt="A screenshot of the Google APIs new project creation window"/></a>
+
+Once you've created the project, click on 'Enable APIs and Services'
+
+<a href="/2020/09/02/Rclone-upload-to-the-cloud-from-cli/rclone_google_api_2-full.png"><img src="/2020/09/02/Rclone-upload-to-the-cloud-from-cli/rclone_google_api_2-full.png" alt="A screenshot of the Google APIs dashboard"/></a>
+
+Search for 'Drive' and click on the Google Drive API, which should be the first option.
+
+<a href="/2020/09/02/Rclone-upload-to-the-cloud-from-cli/rclone_google_api_3-full.png"><img src="/2020/09/02/Rclone-upload-to-the-cloud-from-cli/rclone_google_api_3-full.png" alt="A screenshot of the Google API search"/></a>
+
+Enable the Google Drive API
+
+<a href="/2020/09/02/Rclone-upload-to-the-cloud-from-cli/rclone_google_api_4-full.png"><img src="/2020/09/02/Rclone-upload-to-the-cloud-from-cli/rclone_google_api_4-full.png" alt="A screenshot of the Google Drive API welcome screen"/></a>
+
+Once you have enabled this- Do not click 'Create Credentials' directly. Instead, click on Credentials in the left navigation pane, then 'Create Credentials' from that page.
```

Comment:
Might be nice to explain why not -- looks like Google left an attractive nuisance there. :stuck_out_tongue_winking_eye:

Reply:
"It takes you to the wrong wizard" was too good a line not to use here.